es-cec-input.py

```diff
@@ -78,146 +89,224 @@ def get_keymap():
             'y': uinput.KEY_Y, 'z': uinput.KEY_Z
             }
 
-    return keymap
 
-
-def generate_keylist():
-    """generate a list of keys we actually need
+def generate_controller_to_uinput_codes_map():
+    """generate a map of keys we actually need
     this will be stored in memory and will comprise of
     a,b,x,y,start,select,l,r,left,right,up,down,l2,r2,l3,r3
     keyboard corresponding values the user has chosen
     in the retroarch.cfg file"""
 
-    keylist = []
-    key_bindings = get_key_bindings('/opt/retropie/configs/all/retroarch.cfg')
-    keymap = get_keymap()
+    c_to_u = {}
+    controller_to_keyboard_bindings = get_key_bindings(
+        '/opt/retropie/configs/all/retroarch.cfg')
+    keyboard_to_uinput_mappings = get_keyboard_to_uinput_map()
     errors = []
 
-    for binding in key_bindings:
-
+    for controller_key in controller_to_keyboard_bindings:
+        keyboard_key = controller_to_keyboard_bindings[controller_key]
         try:
-            keylist.append(keymap[binding])
+            c_to_u[controller_key] = keyboard_to_uinput_mappings[keyboard_key]
         except KeyError as e:
             errors.append(e)
 
     if (len(errors) > 0):
         print 'The %s keys in your retroarch.cfg are unsupported\
                 by this script\n' % ', '.join(map(str, errors))
         print 'Supported keys are:\n'
-        print get_keymap().keys()
+        print keyboard_to_uinput_mappings.keys()
         sys.exit()
 
-    return keylist
+    return c_to_u
 
 
 def get_key_bindings(ra_cfg):
     """read key mappings from retroarch.cfg file.
     returns the corresponding keys the user mapped
     in the retroarch.cfg file"""
 
-    keys = []
+    keys = {}
     with open(ra_cfg, 'r') as fp:
         for line in fp:
+<<<<<<< 9ea7e255376d2db0c22efcedfa097b5a640977fd
```

Comment:
There is a conflict starting this line @guicamest

Reply:
Thanks for the heads up @paradadf ! It's been like this for more than a year... :|
Already fixed it and updated :)

Reply:
@guicamest, I wanted to contact you about other things in this improvements you did but couldn’t find a way. Are somewhere on discord or how could I write you a line? (Sorry, I know this isn’t the place)

Reply:
Feel free to comment on the relevant commit: https://github.com/guicamest/es-cec-input/commits/pressed